components/fxa-client/src/device.rs

```diff
@@ -20,12 +20,14 @@ use std::collections::{HashMap, HashSet};
 impl FirefoxAccount {
     /// Fetches the list of devices from the current account including
     /// the current one.
-    pub fn get_devices(&self) -> Result<Vec<Device>> {
+    pub fn get_devices(&mut self) -> Result<Vec<Device>> {
+        let client = self.client.clone();
+        let config = &self.state.config.clone();
         let refresh_token = self.get_refresh_token()?;
-        self.client.devices(&self.state.config, &refresh_token)
+        client.devices(config, &refresh_token)
```

Comment:
@thomcc do you think it's the right fix for https://github.com/mozilla/application-services/pull/2481/files/1b95d9e91f653a30e00ea6aeed354594523af57f#diff-1bde00663b7e4a063fc914419e896b59 ? (we are cloning clients and config everywhere)

Reply:
hm i think we figured it out! the refresh_token was a ref